src/schema/quest.py

```diff
@@ -56,6 +56,11 @@ class Config:
     user_problem_solutions: Optional[List[str]] = Field(
         default_factory=list, description="The user's solutions to the problems."
     )
+
+    current_problem: Optional[str] = Field(
```

Comment:
fwiw, this is going to be a challenge for mixing fixed / generated quests. i think having some richer representation of problems for both paths is going to ultimately required.